### ISSUE TITLE
Support poll participation in stages

### ISSUE THREAD

**Issue by PhrozenByte** [nextcloud/polls#3099]
### ⚠️ This issue respects the following points: ⚠️

- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Describe the goal you'd like to achieve

As a **poll owner**, I want to allow "staged" polls: I first share my poll with participants of group A *only*, then I want to close the poll for people in group A and open the same poll (i.e. including the previous votes of the people in group A) for participants of group B *only*.

The reason is the following: I want to use `polls` for group/team forming (each poll option is a team), but want to ensure that the people in the formed teams consist of different groups of people.

### Describe possible solutions

I was thinking about allowing *read only shares*: A poll owner can set any share as *read only*, which basically causes the poll to look like if it has been closed for this user. With such share one can't edit one's vote, but still sees other votes (unless it's an anonymous poll). It should be possible to set a share read only and back to normal at any time.

Another, but less powerful approach is to utilize that `polls` doesn't delete votes when deleting shares: The poll owner first shares the poll with people of group A only and later deletes all shares of the participants. `polls` won't delete the votes of the participants. The poll owner then shares the poll with people of group B. However, this has two major downsides: Currently the names of poll participants whose share was deleted is replaced by "Deleted user", i.e. the name needs to be retained in the future. Another issue is that the participants of group A don't just loose the ability to vote, but to access the poll at all.

**Comment by dartcafe** [comment]
Something similair is justed checked in to the master #3096

Revoking a share means the user has no more write access to the poll, but can still enter it, if he is involved in the poll (as a voter) and is an internal user. 

![grafik](https://github.com/nextcloud/polls/assets/26707476/221f3fe9-d469-4cf1-afc5-e55b16eb38bf)


But the current implementation denies any access for external users, if the share is revoked. Maybe it can also extended to a read only access. Internal users are still able to see the results.

**Comment by PhrozenByte** [comment]
> Something similair is justed checked in to the master #3096

What a coincidence! Great work! :heart:

> But the current implementation denies any access for external users, if the share is revoked. Maybe it can also extended to a read only access. Internal users are still able to see the results.

Yes, this would be great, as it would indeed fully resolve the suggestion.

Quick question: What happens when the share of an external user is revoked (with #3096 merged)? Does it also delete the vote, or does it replace the name by "Deleted user" (as before), or is just access denied but the vote is kept as-is?

**Comment by dartcafe** [comment]
Votes will stay untouched with the name of the voter.

**Comment by dartcafe** [comment]
I enhanced the feature. published in beta4

**Comment by PhrozenByte** [comment]
I promised to give some feedback about the new feature, but got sick in the meantime, so sorry for the delay. Anyway, it works just perfect! :+1: I was able to do a "staged poll" with beta 5, i.e. let people in group A vote first, then people in group B, resulting in mixed teams - just as I wanted it to work. No problems in production for about 70 poll participants. Thanks again for the super quick implementation! :heart: My next larger poll might take some time, but I'm sure beta 6, rc 1 and the final make the feature even better :relaxed:
